src/Perl6/Grammar.nqp

```diff
@@ -1980,23 +1980,37 @@ grammar Perl6::Grammar is HLL::Grammar does STD {
 
                 # Unless we're augmenting...
                 if $*SCOPE ne 'augment' {
-                    if $longname {
-                        for $longname.colonpairs_hash('class') -> $adverb {
-                            my str $key := $adverb.key;
-                            if $key eq 'ver' {
-                                $*VER := $*W.handle-begin-time-exceptions($/,
-                                    'parsing package version',
-                                    -> { $*W.find_single_symbol_in_setting('Version').new($adverb.value) });
-                            }
-                            elsif $key eq 'api' {
-                                $*API := $adverb.value;
-                            }
-                            elsif $key eq 'auth' {
-                                $*AUTH := $adverb.value;
+                    unless $*COMPILING_CORE_SETTING {
+                        my $Version := $*W.find_single_symbol_in_setting('Version');
+                        if $longname {
+                            for $longname.colonpairs_hash('class') -> $adverb {
+                                my str $key := $adverb.key;
+                                if $key eq 'ver' {
+                                    $*VER := $*W.handle-begin-time-exceptions($/,
+                                        'parsing package version',
+                                        -> { $Version.new($adverb.value) });
+                                }
+                                elsif $key eq 'api' {
```

Comment:
`api` should be a `Version` as we treat it like a version everywhere else (particularly for its sort logic)